pkg/cloudprovider/providers/gce/cloud/gen.go

```diff
@@ -0,0 +1,10446 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was generated by "go run gen/main.go > gen.go". Do not edit
+// directly.
+
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+	"google.golang.org/api/googleapi"
+
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/filter"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/meta"
+
+	alpha "google.golang.org/api/compute/v0.alpha"
+	beta "google.golang.org/api/compute/v0.beta"
+	ga "google.golang.org/api/compute/v1"
+)
+
+// Cloud is an interface for the GCE compute API.
+type Cloud interface {
+	Addresses() Addresses
```

Comment:
Just a thought, maybe it's worth always adding "regional/region" or "zonal/zone" in the function names for consistency (at the cost of slightly longer names). It makes the function names more self-contained by themselves (and not relying on extra knowledge of the GCE resources)

Reply:
Not sure about this -- as the compute API already adds "Global", "Region" in inconsistent ways so it'll look even more strange (GlobalAddresses, RegionDisk, ...)